addons/mrp/static/src/product_catalog/kanban_record.js

```diff
@@ -0,0 +1,17 @@
+/** @odoo-module */
+import { ProductCatalogKanbanRecord } from "@product/product_catalog/kanban_record";
+import { ProductCatalogMrpMoveRawLine } from "./mrp_move_raw/mrp_move_raw";
+import { patch } from "@web/core/utils/patch";
+
+patch(ProductCatalogKanbanRecord.prototype, {
```

Comment:
extend new class

Reply:
Hello @amoyaux,

Since 'ProductCatalogKanbanRecord' extends 'KanbanRecord', So it's increase the complexity and also line of code, So i think it's better to patch it rather than extend it in a new class. 

A similar approach has been applied in the purchase module.
https://github.com/odoo/odoo/blob/master/addons/purchase/static/src/product_catalog/kanban_record.js#L9

Thanks.

Reply:
50 lines of javascript to replace a shopping icon by a gear icon is not acceptable

Reply:
I'm sure than adding a new props for the icon is a better idea